step-automation-packages/step-automation-packages-schema/src/main/java/step/automation/packages/yaml/schema/YamlKeywordSchemaGenerator.java

```diff
@@ -50,30 +49,29 @@ public YamlKeywordSchemaGenerator(JsonProvider jsonProvider) {
         this.schemaHelper = new YamlJsonSchemaHelper(jsonProvider);
 
         // --- Fields metadata rules (fields we want to rename)
-        FieldMetadataExtractor fieldMetadataExtractor = prepareMetadataExtractor();
+        FieldMetadataExtractor fieldMetadataExtractor = new DefaultFieldMetadataExtractor();
 
         List<JsonSchemaFieldProcessor> processingRules = prepareFieldProcessors();
         this.jsonSchemaCreator = new JsonSchemaCreator(jsonProvider, new AggregatedJsonSchemaFieldProcessor(processingRules), fieldMetadataExtractor);
     }
 
+    protected List<JsonSchemaDefinitionCreator> getDefinitionsExtensions() {
```

Comment:
@jeromecomte 
The idea looks good, but we need to discuss how to do it in the best way. As I see, there are 2 separate issues:

1. The json schema generation. 
In fact, these scanned `JsonSchemaDefinitionCreators` are only used in `YamlPlanJsonSchemaGenerator` and `YamlAutomationPackageSchemaGenerator`. The generation is called from standalone tools (`YamlPlanSchemaGenerationTool`) and doesn't use plugins engine.  So I'm not sure that plugins is the best place to register such extensions for json schema.  Alternatively to avoid scanning we could simply register all creators manually inside the `YamlPlanJsonSchemaGenerator`, but in this case we should not forget to do this every time when we need to extend the schema. So I still don't know, which approach is better...

2. The extensions for runtime (like custom serializers/deserializers), which really could be registered in plugins. But there are some very common serializers (like `YamlDynamicValueSerializer`), which should be always used (explicitly listed in `YamlPlanReader`?). Also we need to register these extensions not only in controller plugins, but also for junit runners and in tests.

FYI. I've checked the slowest annotation scans in Step and found the following:
* the `ExecutionEngine.withPluginsFromClasspath()` used in junit runner
* the `AnnotationScanner.forAllClassesFromClassLoader(archive.getClassLoader())` in `AutomationPackageReader` used to scan keywords and plans
Other ones (like for `StepYamlDeserializerAddOn`) are much faster, because they are restricted with 'step' package and don't scan the whole classpath.

Reply:
@iegorov777 Thanks for the analysis, I'll created a dedicated ticket for next release to see if there are any improvement to be done. For this iterations, we can keep the implementation as it is.